### PR TITLE
supported cypress for team and glossary drag and drop functionality

### LINKED ISSUE #14893: Add Cypress test for Drag and Drop functionality
- [ ]  Teams Page
- [ ] Glossary Page

### PR DESCRIPTION
**Pull request by Ashish8689**
### Describe your changes:

Fixes #14893

### Team
- added cypress for team drag and drop functionality
- added droppable style in the table while dragging.
- supported parent team level drop, through dropping on table header.
- added restrictions on team transfer based on team type

Restricted Drop : 

- Any type drop-in Group is not allowed
- BusinessUnit drop-in Division is not allowed
- BusinessUnit drop-in Department is not allowed
- Division drop-in Department is not allowed

### Glossary : 
- Term level drag and drop
- Table level drag and drop

### SearchIndexApplication
- Also fixes SearchIndexApplication cypress failure.
  

https://github.com/open-metadata/OpenMetadata/assets/66266464/a5c09c76-ec66-4510-b0ec-b7f102430b5e

https://github.com/open-metadata/OpenMetadata/assets/66266464/3fc008f9-23ae-4703-80d2-2afd8284cc0a

#
### Type of change:

- [ ] Bug fix
- [x] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/cypress/common/common.js [hunk 1]
```diff
@@ -1123,3 +1123,32 @@ export const visitDatabaseSchemaDetailsPage = ({
     .contains(databaseSchemaName)
     .click();
 };
+
+const dataTransfer = new DataTransfer();
+
+export const dragAndDropElement = (dragElement, dropTarget, isHeader) => {
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/cypress/common/common.js:1129]: Let's not add anything to common js
We can have DragAndDrop.ts instead in the utils

File: openmetadata-ui/src/main/resources/ui/src/utils/TeamUtils.ts [hunk 2]
```diff
@@ -126,3 +129,21 @@ export const getTeamOptionsFromType = (parentType: TeamType) => {
...
 };
+
+/**
+ * Restricting the drop of team based on the team type
+ * Group: Can't have any child team
+ * Division: Can have only Department and Group
+ * Department: Can have only Group
+ */
+
+export const isDropRestricted = (
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/utils/TeamUtils.ts:140]: Can you add unit test for this?

File: openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryTermTab/GlossaryTermTab.component.tsx [hunk 3]
```diff
@@ -426,17 +427,19 @@ const GlossaryTermTab = ({
...
-            renderElement={<strong />}
-            values={{
-              from: movedGlossaryTerm?.from.name,
-              to: movedGlossaryTerm?.to?.name ?? getEntityName(selectedData),
-              entity: isUndefined(movedGlossaryTerm?.to)
-                ? ''
-                : t('label.term-lowercase'),
-            }}
-          />
+          <Typography.Text data-testid="transfer-message">
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/Glossary/GlossaryTermTab/GlossaryTermTab.component.tsx:430]: Not sure if we need an element wrapper just for test id
Can we avoid it

**Comment by Ashish8689** [comment]
@chirag-madlani changes made as per comments!

**Review by chirag-madlani — APPROVED** [review]

**Review by chirag-madlani — APPROVED** [review]

**Pull request merged**